### PR TITLE
feat(java-binding): support setting vnodes to read in java-binding read plan

### PR DESCRIPTION
**Pull request by wenym1**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

As titled

## Checklist For Contributors

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [x] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Checklist For Reviewers

- [x] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

## Documentation

- [x] My PR **DOES NOT** contain user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

File: src/java_binding/src/lib.rs [hunk 1]
```diff
@@ -207,6 +209,16 @@ where
     }
 }
 
+// JNIEXPORT jint JNICALL Java_com_risingwave_java_binding_Binding_vnodeCount
+// (JNIEnv *, jclass);
```
Comment (Gun9niR) [inline src/java_binding/src/lib.rs:213]: Can this be removed?

File: src/java_binding/src/lib.rs [hunk 2]
```diff
@@ -207,6 +209,16 @@ where
...
+
+#[no_mangle]
+pub extern "system" fn Java_com_risingwave_java_binding_Binding_vnodeCount(
+    _env: EnvParam<'_>,
+) -> jint {
+    VirtualNode::COUNT as jint
```
Comment (Gun9niR) [inline src/java_binding/src/lib.rs:219]: Is this a temporary value? I would imagine multiple `Iterator` instances reading the same table, so each iter should read a subset of all vnodes?

Comment (wenym1) [reply src/java_binding/src/lib.rs:216]: This is for returning the total number of vnode defined in RisingWave, which is a const value defined in our system.

In the upper level query engine, we will get the total vnode count via this call and assign different subsets of vnodes to different parallelism.

File: src/java_binding/src/lib.rs (hunk 1 above)
Comment (wenym1) [reply src/java_binding/src/lib.rs]: fixed.

**Review by Gun9niR — APPROVED** [review]

**Pull request merged**